overlord/devicestate/handlers.go

```diff
@@ -63,13 +68,54 @@ func deviceAPIBaseURL() string {
 }
 
 var (
-	keyLength        = 4096
-	retryInterval    = 60 * time.Second
-	deviceAPIBase    = deviceAPIBaseURL()
-	requestIDURL     = deviceAPIBase + "request-id"
-	serialRequestURL = deviceAPIBase + "devices"
+	keyLength        uint64 = 4096
+	retryInterval           = 60 * time.Second
+	deviceAPIBase           = deviceAPIBaseURL()
+	requestIDURL            = deviceAPIBase + "request-id"
+	serialRequestURL        = deviceAPIBase + "devices"
+	sshKeyFile              = "/tmp/snapd.key.tmp"
+	sshPublicKeyFile        = sshKeyFile + ".pub"
 )
 
+func sshGenerateKey() (*rsa.PrivateKey, error) {
```

Comment:
I did a couple of test runs and both master and this branch take mostly the same time, differences are in the range of milliseconds.